spec/ffi/callback_spec.rb

```diff
@@ -771,3 +771,96 @@ module LibTestStdcall
     end
   end
 end
+
+describe "Callback interop" do
+  require 'fiddle'
+  require 'fiddle/import'
+  require 'timeout'
+
+  module LibTestFFI
+    extend FFI::Library
+    ffi_lib TestLibrary::PATH
+    attach_function :testCallbackVrV, :testClosureVrV, [ :pointer ], :void
+    attach_function :testCallbackVrV_blocking, :testClosureVrV, [ :pointer ], :void, blocking: true
+  end
+
+  module LibTestFiddle
+    extend Fiddle::Importer
+    dlload TestLibrary::PATH
+    extern 'void testClosureVrV(void *fp)'
+  end
+
+  def assert_callback_in_same_thread_called_once
+    called = 0
+    thread = nil
+    yield proc {
+      called += 1
+      thread = Thread.current
+    }
+    expect(called).to eq(1)
+    expect(thread).to eq(Thread.current)
+  end
+
+  it "from ffi to ffi" do
+    assert_callback_in_same_thread_called_once do |block|
+      func = FFI::Function.new(:void, [:pointer], &block)
+      LibTestFFI.testCallbackVrV(FFI::Pointer.new(func.to_i))
+    end
+  end
+
+  it "from ffi to ffi with blocking:true" do
+    assert_callback_in_same_thread_called_once do |block|
+      func = FFI::Function.new(:void, [:pointer], &block)
+      LibTestFFI.testCallbackVrV_blocking(FFI::Pointer.new(func.to_i))
+    end
+  end
+
+  # https://github.com/ffi/ffi/issues/527
+  if RUBY_VERSION.split('.').map(&:to_i).pack("C*") >= [2,3,0].pack("C*") || RUBY_PLATFORM =~ /java/
+    it "from fiddle to ffi" do
+      assert_callback_in_same_thread_called_once do |block|
+        func = FFI::Function.new(:void, [:pointer], &block)
+        LibTestFiddle.testClosureVrV(Fiddle::Pointer[func.to_i])
+      end
+    end
+  end
+
+  it "from ffi to fiddle" do
+    assert_callback_in_same_thread_called_once do |block|
+      func = LibTestFiddle.bind_function(:cbVrV, Fiddle::TYPE_VOID, [], &block)
+      LibTestFFI.testCallbackVrV(FFI::Pointer.new(func.to_i))
+    end
+  end
+
+  it "from ffi to fiddle with blocking:true" do
+    assert_callback_in_same_thread_called_once do |block|
+      func = LibTestFiddle.bind_function(:cbVrV, Fiddle::TYPE_VOID, [], &block)
+      LibTestFFI.testCallbackVrV_blocking(FFI::Pointer.new(func.to_i))
+    end
+  end
+
+  it "from fiddle to fiddle" do
+    assert_callback_in_same_thread_called_once do |block|
+      func = LibTestFiddle.bind_function(:cbVrV, Fiddle::TYPE_VOID, [], &block)
+      LibTestFiddle.testClosureVrV(Fiddle::Pointer[func.to_i])
+    end
+  end
+
+  # https://github.com/ffi/ffi/issues/527
+  if RUBY_ENGINE == 'ruby' && RUBY_VERSION.split('.').map(&:to_i).pack("C*") >= [2,3,0].pack("C*")
```

Comment:
I don't understand your last comment (as a non English speaker). Shall I adjust the tests so that they pass on rbx (although it is no longer tested on travis-ci)?

Anyway I already tested this with JRuby (see https://github.com/ffi/ffi/pull/584/commits/047371bd21cec9780c53f563d6661e3e65b4fd9a) , so that the update of the testsuite could be merged there someday.